network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -208,19 +208,38 @@ func (c *ControlMsgValidationInspector) processInspectRPCReq(req *InspectRPCRequ
 // - activeClusterIDS: the list of active cluster ids.
 // Returns:
 // - DuplicateTopicErr: if there are any duplicate topics in the list of grafts
-func (c *ControlMsgValidationInspector) inspectGraftMessages(from peer.ID, grafts []*pubsub_pb.ControlGraft, activeClusterIDS flow.ChainIDList) error {
+func (c *ControlMsgValidationInspector) inspectGraftMessages(from peer.ID, grafts []*pubsub_pb.ControlGraft, activeClusterIDS flow.ChainIDList) p2p.InvCtrlMsgErrs {
+	lg := c.logger.With().
+		Str("peer_id", p2plogging.PeerId(from)).
+		Int("sample_size", len(grafts)).
+		Logger()
 	tracker := make(duplicateStrTracker)
+	errs := make(p2p.InvCtrlMsgErrs, 0)
 	for _, graft := range grafts {
 		topic := channels.Topic(graft.GetTopicID())
 		if tracker.isDuplicate(topic.String()) {
-			return NewDuplicateTopicErr(topic.String(), p2pmsg.CtrlMsgGraft)
+			errs = append(errs, p2p.NewInvCtrlMsgErr(NewDuplicateTopicErr(topic.String(), p2pmsg.CtrlMsgGraft), p2p.ModerateErrSeverity))
+			continue
 		}
 		tracker.set(topic.String())
 		err := c.validateTopic(from, topic, activeClusterIDS)
 		if err != nil {
-			return err
+			// short circuit validation when unstaked peer detected
+			if IsErrUnstakedPeer(err) {
+				return nil
+			}
+			errs = append(errs, p2p.NewInvCtrlMsgErr(err, p2p.HighErrSeverity))
 		}
 	}
+
+	lg.Debug().
+		Int("error_count", errs.Len()).
+		Msg("graft control message validation complete")
```

Comment:
When `errs.Len() > 0` we need to add a `suspicious:true` key to the log. This also applies to other message types.

Reply:
https://github.com/onflow/flow-go/pull/4978/commits/8aa7ef853b8bc100793c05bc728b765df849c86f